clients/src/main/java/org/apache/kafka/common/MetricName.java

```diff
@@ -87,11 +87,13 @@ public MetricName(String name, String group, String description, String... keyVa
 
     private static Map<String, String> getTags(String... keyValue) {
         if ((keyValue.length % 2) != 0)
-            throw new IllegalArgumentException("keyValue needs to be specified in paris");
+            throw new IllegalArgumentException("keyValue needs to be specified in pairs");
         Map<String, String> tags = new HashMap<String, String>();
 
-        for (int i = 0; i < keyValue.length / 2; i++)
+        for (int i = 0; i < keyValue.length / 2;) {
```

Comment:
@guozhangwang No need to divide by 2 in the for-loop as it hops by 2 each iteration, so when it reaches ``keyValue.length - 2`` the next value of ``i`` will be keyValue.length. ;)